docs/freefeatures.rst

```diff
@@ -253,9 +253,198 @@ the readline history.
 
 .. automethod:: cmd2.cmd2.Cmd.__init__
 
-``cmd2`` makes a third type of history access available with the **history** command:
+``cmd2`` makes a third type of history access available with the `history` command. Each time
+the user enters a command, ``cmd2`` saves the input. The `history` command lets you do interesting
+things with that saved input. The examples to follow all assume that you have entered the
+following commands::
 
-.. automethod:: cmd2.cmd2.Cmd.do_history
+    (Cmd) alias create one !echo one
+    Alias 'one' created
+    (Cmd) alias create two !echo two
+    Alias 'two' created
+    (Cmd) alias create three !echo three
+    Alias 'three' created
+    (Cmd) alias create four !echo four
+    Alias 'four' created
+
+In it's simplest form, the `history` command displays previously entered commands. With no
+additional arguments, it displays all previously entered commands::
+
+    (Cmd) history
+        1  alias create one !echo one
+        2  alias create two !echo two
+        3  alias create three !echo three
+        4  alias create four !echo four
+
+If you give a positive integer as an argument, then it only displays the specified command::
+
+    (Cmd) history 4
+        4  alias create four !echo four
+
+If you give a negative integer *N* as an argument, then it display the *Nth* last command.
+For example, if you give `-1` it will display the last command you entered. If you give `-2`
+it will display the next to last command you entered, and so forth::
+
+    (Cmd) history -2
+        3  alias create three !echo three
+
+You can use a similar mechanism to display a range of commands. Simply give two command numbers
+separated by `..` or `:`, and you will see all commands between those two numbers::
+
+    (Cmd) history 2:3
+        2  alias create two !echo two
+        3  alias create three !echo three
+
+If you omit the first number, it will start at the beginning. If you omit the last number, it
+will continue to the end::
+
+    (Cmd) history :2
+        1  alias create one !echo one
+        2  alias create two !echo two
+    (Cmd) history 2:
+        2  alias create two !echo two
+        3  alias create three !echo three
+        4  alias create four !echo four
+
+You can use negative numbers as either the first or second number of the range (but not both). If
+you want to display the last three commands entered::
+
+    (Cmd) history -- -3:
+        2  alias create two !echo two
+        3  alias create three !echo three
+        4  alias create four !echo four
+
+Notice the double dashes. These are required because the history command uses `argparse` to parse
```

Comment:
Ah, good documentation.  It explains the weirdness I was experiencing.  I guess it is treating the "-" as a dash and attempting to match a flag.

Reply:
@kotfu @tleonhardt 
This is the official description of '--' from [argparse documentation](https://docs.python.org/3/library/argparse.html).

> If you have positional arguments that must begin with - and don’t look like negative numbers, you can insert the pseudo-argument '--' which tells parse_args() that everything after that is a positional argument

Maybe we can put a summary of this in freefeatures.rst.

Reply:
Added.